v-next/hardhat/src/internal/builtin-plugins/network-manager/type-extensions/config.ts

```diff
@@ -33,16 +33,67 @@ declare module "../../../../types/config.js" {
     httpHeaders?: Record<string, string>;
   }
 
+  export type IntervalMiningConfig = number | [number, number];
+
+  export type MempoolOrder = "fifo" | "priority";
+
+  export type HardforkHistoryConfig = Map<
+    /* hardforkName */ string,
+    /* blockNumber */ number
+  >;
+
+  export interface HardhatNetworkChainConfig {
+    hardforkHistory: HardforkHistoryConfig;
+  }
+
+  export type HardhatNetworkChainsConfig = Map<
+    /* chainId */ number,
+    HardhatNetworkChainConfig
+  >;
+
+  export interface GenesisAccount {
+    privateKey: string;
+    balance: string | number | bigint;
+  }
+
+  export interface ForkConfig {
+    jsonRpcUrl: string;
+    blockNumber?: bigint;
+    httpHeaders?: Record<string, string>;
+  }
```

Comment:
We are creating the equivalent of EDR types here to avoid putting EDR types directly in the public API.